packages/xo-server/src/api/xostor.mjs

```diff
@@ -66,53 +76,57 @@ updateDependencies.resolve = {
 
 export async function formatDisks({ disks, force, host, ignoreFileSystems, provisioning }) {
   const rawDisks = disks.join(',')
-  const xapi = this.getXapi(host)
-
-  const lvmPlugin = (fnName, args) => pluginCall(xapi, host, 'lvm.py', fnName, args)
-  log.info(`Format disks (${rawDisks}) with force: ${force}`, { hostId: host.id })
-
-  if (force) {
-    await destroyVolumeGroup(xapi, host, force)
-  }
-
-  // ATM we are unable to correctly identify errors (error.code can be used for multiple errors.)
-  // so we are just adding some suggestion of "why there is this error"
-  // Error handling will be improved as errors are discovered and understood
-  try {
-    await lvmPlugin('create_physical_volume', {
-      devices: rawDisks,
-      ignore_existing_filesystems: String(ignoreFileSystems),
-      force: String(force),
-    })
-  } catch (error) {
-    if (error.code === 'LVM_ERROR(5)') {
-      error.params = error.params.concat([
-        "[XO] This error can be triggered if one of the disks is a 'tapdevs' disk.",
-        '[XO] This error can be triggered if one of the disks have children',
-      ])
-    }
-    throw error
-  }
-  try {
-    await lvmPlugin('create_volume_group', {
-      devices: rawDisks,
-      vg_name: VG_NAME,
-    })
-  } catch (error) {
-    if (error.code === 'LVM_ERROR(5)') {
-      error.params = error.params.concat([
-        "[XO] This error can be triggered if a VG 'linstor_group' is already present on the host.",
-      ])
-    }
-    throw error
-  }
+  return Task.run(
+    { properties: { disks, name: `format disks on ${host.name_label}`, objectId: host.uuid } },
+    async () => {
+      const xapi = this.getXapi(host)
 
-  if (provisioning === ENUM_PROVISIONING.Thin) {
-    await lvmPlugin('create_thin_pool', {
-      lv_name: LV_NAME,
-      vg_name: VG_NAME,
-    })
-  }
+      const lvmPlugin = (fnName, args) => pluginCall(xapi, host, 'lvm.py', fnName, args)
+
+      if (force) {
+        await destroyVolumeGroup(xapi, host, force)
```

Comment:
here too, I would put all the different steps in theri own sub task, that way the use will see the progress even without a progress bar, and error will be easy to contextualize